code/modules/hydroponics/beekeeping/beebox.dm

```diff
@@ -71,8 +71,12 @@
 
 	var/datum/reagent/R = null
 	if(random_reagent)
-		R = pick(subtypesof(/datum/reagent))
-		R = chemical_reagents_list[initial(R.id)]
+		var/synthable = FALSE
+		while(!synthable)
+			R = pick(subtypesof(/datum/reagent))
+			if(R.can_synth)
+				R = chemical_reagents_list[initial(R.id)]
+				synthable = TRUE
```

Comment:
You can condense this entire thing by just using `R = get_random_reagent_id()` which already checks against can_synth

Reply:
That's helpful!